contracts/strategies/88mph/base/EightyEightMphStrategy.sol

```diff
@@ -0,0 +1,676 @@
+        // Fail if confirm flag is not passed in as true to ensure whoever triggers this understands that it must be arranged 
+        // for the withdrawal fee to be waived for the new deposit Id
+        require(confirmWithdrawalFeeWillBeWaived, "Withdrawal fee arrangement confirmation missing");
+
+        // get current deposit data
+        uint256 virtualTokenTotalSupply;
+        uint256 interestRate;
+        uint64 depositMaturity;
+        (virtualTokenTotalSupply, interestRate,,, depositMaturity,) = IDInterest(rewardPool()).getDeposit(depositId);
+
+        // ensure deposit has reached maturity to get the fixed yield rate
+        // alternatively if the deposit has no funds, the fixed yield rate doesn't matter anyway because we create a new deposit
+        require(virtualTokenTotalSupply == 0 || block.timestamp > depositMaturity, "Deposit has not reached maturity yet");
+
+        // 88mph does not support rolling over deposits that have no funds.
+        // if this case really comes to happen then this would have to be resolved manually
+        // Note that for this to happen, all depositors would have to withdraw everything and the
+        // strategy would not have any active users or everything is deposited in the vault (withdrawAllToVault)
+        // if the strategy at least has some underlying balance we can create a new deposit instead of rolling over the old one.
+        // if neither of both is true, then calling rolloverDeposit should revert
+        uint256 underlyingBalance = IERC20(underlying()).balanceOf(address(this));
+        require(virtualTokenTotalSupply > 0 || underlyingBalance > 0, "Deposit has no funds, rollover not supported");
+
+        if(virtualTokenTotalSupply <= 0) {
+             // ---------- create new deposit because old deposit is drained ----------
+            // empty deposit. fixed yield is not available anyway, best we can do here is create a new deposit
+            // which has essentially the same effect as rolling over a deposit
+            // note that we can not simply top up the old deposit, since 88mph logic doesn't support that
+            // thanks to the require check a few lines up we know that we have underlying balance available.
+            createNewDeposit();
+        } else {
+            // ------------------------ actual rollover Deposit --------------------
+            _rolloverDeposit(virtualTokenTotalSupply, interestRate);
+        }
+
+        // reset flag
+        shouldRolloverDeposit = false;
+    }
+
+    function setPotPool(address _value) public onlyGovernance {
+        require(potPool() == address(0), "PotPool already set");
+        setAddress(_POTPOOL_SLOT, _value);
+    }
+
+    function potPool() public view returns (address) {
+        return getAddress(_POTPOOL_SLOT);
+    }
+
+    /**
+     * Sets the percentage for the amount of rewards that will be distributed as xMPH to depositors (after fees)
```

Comment:
For this strategy, per default we 
take 30% as profitSharingFee
reinvest 35% as underlying
stake 35% as xMPH and distribute it through the potPool

The amount that is staked as xMPH can be adjusted